BDD100K : Test Files

ananno opened the pull request:
<!--- 👉👉👉 Provide a general summary of your changes. 👈👈👈 -->

## Motivation and Context
<!--- 💭💡💭 Why is this change required? What problem does it solve? 💭💡💭 -->
<!--- 🔗 If it fixes an open issue, please link to the issue here. 🔗 -->
Adding unit testing file for bdd100k.

## Description
<!--- 📝🎯📝 Describe your changes in detail. 📝🎯📝 -->
To automatic unit testing of both object detection and segmentation dataset classes this test files is implemented following blueoil test framework.

## How has this been tested?
<!--- 🙆👌🙆 Please describe in detail how you tested your changes. 🙆👌🙆 -->
<!--- Include details of your testing environment, details of your manual tests, snippet code or commands of your manual tests, table of experiments results metrics, tests ran to see how your change affects other areas of the code, etc. -->
<!--- If you have experiments report documents please link to here. -->
`tox -e pytest`

## Screenshots (if appropriate):

## Types of changes
<!--- 🏁🎌🏁 What types of changes does your code introduce? Put an `x` in all the boxes that apply: 🏁🎌🏁 -->
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature / Optimization (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist:
<!--- ✅✅✅ Go over all the following points, and put an `x` in all the boxes that apply. ✅✅✅ -->
<!--- If you're unsure about any of these, don't hesitate to ask. We're here to help! -->

<!--- TODO(wakisaka): After decide our code style, add this.
- [ ] My code follows the code style of this project.
-->
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.

ananno commented:
@iizukak San:
I began to separate #399 into multiple smaller PRs. This is the 5th one. Please review.

ananno commented:
@tkng, @iizukak San:
I've applied isort as requested. Please review one more time.

iizukak commented:
@ananno Please solve the conflicts.

ananno commented:
@iizukak San: Can you review this one more time.おねがいします

iizukak commented:
@ananno 
Please discuss about fixture image size in this issue.
https://github.com/blue-oil/blueoil/issues/700

ananno commented:
Commit #796f705 fixes Issue #700

ananno commented:
> @ananno
> Why both `100.jpg` and `1000.png` are exist?

@iizukak San:
You were right. We actually don't need 2 image files. I've removed one and use links to one image file `dummy_image` where necessary.

Now the fixture directory `BDD100K` becomes `72KB`. 😄 

> NOTE : I've tested with a 0 byte file but it didn't work as `PIL` package actually try to open the image.

iizukak commented:
@ananno 
I think `BDD100K/dummy_image` need filename extension...

iizukak commented:
@ananno ping

ananno commented:
/ready

ananno commented:
@iizukak San:
I've updated the filename for `dummy_image` and also updated the corresponding linked files paths. Please review one more time.
N.B. : The automatic test failed due to no space left in the device.

tsawada commented on lmnet/tests/lmnet_tests/datasets_tests/test_bdd100k.py:
```diff
@@ -0,0 +1,81 @@
+    assert test_image_files.shape[0] == batch_size
+    assert test_label_files.shape[0] == batch_size
+
+    print("file annotations are ready")
```
Don't print anything within tests

tsawada commented on lmnet/tests/lmnet_tests/datasets_tests/test_bdd100k.py:
```diff
@@ -0,0 +1,81 @@
+    assert test_image_files.shape[0] == batch_size
+    assert test_label_files.shape[0] == batch_size
+
+    print("file annotations are ready")
```
same

tsawada commented on lmnet/tests/lmnet_tests/datasets_tests/test_bdd100k.py:
```diff
@@ -0,0 +1,81 @@
+    print("file annotations are ready")
+
+
+if __name__ == '__main__':
```
I think we shouldn't prepare any main function within `pytest` test scripts. might be wrong.

ananno commented:
/ready
